client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/design-picker-design-title.tsx

```diff
@@ -30,10 +44,20 @@ const DesignPickerDesignTitle: FC< Props > = ( { designTitle, selectedDesign } )
 		)
 	);
 
-	const bundleSettings = useBundleSettingsByTheme( selectedDesign.slug );
-
 	let badge: React.ReactNode = null;
-	if ( selectedDesign.software_sets && selectedDesign.software_sets.length > 0 ) {
+	if ( isEnabled( 'themes/tiers' ) ) {
+		const isDefaultVariation = isDefaultGlobalStylesVariationSlug( selectedStyleVariation?.slug );
+		const isLockedStyleVariation =
+			( ! selectedDesign.is_premium && shouldLimitGlobalStyles && ! isDefaultVariation ) ?? false;
+
+		badge = (
+			<ThemeTierBadge
+				className="design-picker-design-title__theme-tier-badge"
+				isLockedStyleVariation={ isLockedStyleVariation }
```

Comment:
I don't think we need the `isLockedStyleVariation` prop. As noted in p1705588567980119-slack-C04DZ8M0GHW, the locked style variations should be highlighted with a different section. The upgrade badge that is displayed above the theme name should be only visible for premium themes.

Reply:
> So I think it makes sense to have the upgrade badge, don't you think?

As long as it is in a separate section as it used to be (see the badge next to "Style Variation":

<img width="315" alt="Screenshot 2024-01-18 at 15 33 44" src="https://github.com/Automattic/wp-calypso/assets/1233880/ee96f8f9-0717-41fc-b1e4-55e781901297">

Otherwise it's not clear that only the default style variation is free:

<img width="320" alt="Screenshot 2024-01-18 at 15 34 05" src="https://github.com/Automattic/wp-calypso/assets/1233880/b879a6ad-6519-4977-801e-e2b96333abad">

Reply:
So for now, I think we can hide the badge for free themes (even if a locked style variation is selected) and I'll work in bringing back the separate section.

Reply:
Set it to `false` on https://github.com/Automattic/wp-calypso/pull/86582/commits/d95f058f2e540bd82235158afb9358dbdd62b65d